lib/MT/Util.pm

```diff
@@ -678,9 +678,20 @@ sub rich_text_transform {
 sub html_text_transform {
     my $str = shift;
     $str = '' unless defined $str;
-    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr)!;
+    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr|script|style)!;
+    my $special_tags = qr!(?:script|style|pre|code)!;
     my @paras = split /\r?\n\r?\n/, $str;
+    my $guard;
     for my $i ( 0 .. @paras - 1 ) {
+        if ( $paras[$i] =~ m!^<($special_tags)! ) {
```

Comment:
```suggestion
        if ( !$guard && $paras[$i] =~ m!^<($special_tags)! ) {
```

I think we should check the outermost special tag.
```pre
<pre>

<code> // should ignore an inside tag
alert(1);
</code>

</pre>
```

Reply:
Thanks! Updated.